### PR TITLE
Add "junit5-java9-engine" sample

### PR DESCRIPTION
**Pull request by sormuras**
## Overview

The PR introduces a very simple, single-test and hard-coded "test engine" demonstrating how to register an engine on the module-path. This sample also uses the version and artifact information retrieval methods introduced on the JUnit Platform 1.1-SNAPSHOT via https://github.com/junit-team/junit5/commit/cd94f9a77945594c28d1a3a6e9fefc3ff3981d22

See https://travis-ci.org/junit-team/junit5-samples/jobs/279804539 for the log output:
```
[...]
:testIceCreamMachine
╷
└─ Ice Cream Machine 47.11 (ice.cream) ✔
   ├─ Stracciatella ✔
   ├─ Chocolate ✔
   └─ Vanilla ✔

Test run finished after 5 ms
[         1 containers found      ]
[         0 containers skipped    ]
[         1 containers started    ]
[         0 containers aborted    ]
[         1 containers successful ]
[         0 containers failed     ]
[         3 tests found           ]
[         0 tests skipped         ]
[         3 tests started         ]
[         0 tests aborted         ]
[         3 tests successful      ]
[         0 tests failed          ]
```

---
I hereby agree to the terms of the JUnit Contributor License Agreement.

### CODE REVIEW

**Comment by sbrannen** [comment]
I assume the examples are about "ice cream" instead of "ice" (German: _Eis_).

So I'd recommend renaming everything accordingly for the non-German speaking members of the community. 😉

File: junit5-java9-engine/src/main/java/ice/cream/Machine.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+import org.junit.platform.engine.ExecutionRequest;
+import org.junit.platform.engine.TestDescriptor;
+import org.junit.platform.engine.TestEngine;
+import org.junit.platform.engine.TestExecutionResult;
+import org.junit.platform.engine.UniqueId;
+import org.junit.platform.engine.support.descriptor.EngineDescriptor;
+
+public class Machine implements TestEngine {
+
+  private FlavorDescriptor sample;
```
Comment (marcphilipp) [inline junit5-java9-engine/src/main/java/ice/cream/Machine.java]: We should not encourage engine authors to store descriptors in instance variables.

Comment (sormuras) [reply junit5-java9-engine/src/main/java/ice/cream/Machine.java]: True. Replaced by generating 3 flavors while discovering and "parsing" them back when executing.

**Comment by sbrannen** [comment]
Quick question...

What kind of indentation are you using? 2 spaces?

Please use the same indentation as the rest of the JUnit 5 projects. 😉

**Comment by sbrannen** [comment]
Otherwise, I think the PR looks pretty good.

**Comment by sbrannen** [comment]
Are you also going to add a link and description in the root README file?

**Comment by sormuras** [comment]
Yes.

And add a small README file to this sample as well?

**Comment by sbrannen** [comment]
+2

**Pull request merged**